finagle-core/src/main/scala/com/twitter/finagle/ssl/KeyCredentials.scala

```diff
@@ -48,4 +50,17 @@ object KeyCredentials {
    */
   case class CertKeyAndChain(certificateFile: File, keyFile: File, caCertificateFile: File)
       extends KeyCredentials
+
+  /**
+    * Indicates that this certificate and key should be used by the
+    * engine factory.
+    *
+    * @param key a PKCS#8 private key. It should not require a password.
+    *
+    * @param keyCertChain the X.509 certificate chain
+    */
+  case class KeyAndCertChain(
+      key: PrivateKey,
+      keyCertChain: Array[X509Certificate])
+    extends KeyCredentials
```

Comment:
Undo this change please. If you would like to add this outside the context of this pull request, we can discuss it in a separate PR.